[Doc] Completed cover-to-cover read-through in ToC order. Lots of changes (#9360)
yugabyte/yugabyte-db: docs/content/latest/api/ysql/datatypes/type_datetime/date-time-data-types-semantics/type-interval/interval-arithmetic/_index.md

---

This section uses the term "moment" as an umbrella for a _timestamptz_ value, a _timestamp_ value, or a _time_ value. (In a broader scenario, a _date_ value is also a moment. But you get an _integer_ value when you subtract one _date_ value from another. And you cannot add or subtract an _interval_ value to/from a _date_ value.) The term "_interval_ arithmetic" is used somewhat loosely to denote these three distinct scenarios:

- **[The interval-interval overload of the "=" operator](#the-interval-interval-overload-of-the-operator)**—_interval-interval_ equality. This is what the term implies.

- **[Interval-only addition/subtraction and multiplication/division](#interval-only-addition-subtraction-and-multiplication-division):** This has two subtopics:

  - _first_ [The _interval-interval_ overloads of the "+" and "-" operators](#the-interval-interval-overloads-of-the-and-operators) to produce a new _interval_ value.

  - And _second_ [The _interval_-number overloads of the "*" and "/" operators](#the-interval-number-overloads-of-the-and-operators) to produce a new _interval_ value.

- **[moment-interval arithmetic](#moment-interval-arithmetic):** This has two subtopics:

>   - _first_ [The moment-moment overloads of the "-" operator](#the-moment-moment-overloads-of-the-operator) to produce an _interval_ value. (Addition of two moments is meaningless and is therefore illegal. So is multiplication or division of a moment by a number.)

Reviewer: ```suggestion
  - _First_ [The moment-moment overloads of the "-" operator](#the-moment-moment-overloads-of-the-operator) to produce an _interval_ value. (Addition of two moments is meaningless and is therefore illegal. So is multiplication or division of a moment by a number.)
```

The PostgreSQL documentation does not carefully specify the semantics of _interval_ arithmetic. This page and its children aim to specify the operations in terms of the individual fields of the [internal representation](../interval-representation/).

> {{< note title="The results of 'interval' arithmetic are, in general, sensitive to the session's timezone." >}}

Reviewer: Straight quotes are almost never correct in text, and in this case you can just dispense with the quotes entirely.
```suggestion
{{< note title="The results of interval arithmetic are, in general, sensitive to the session's timezone." >}}
```

select ('1 mon 30 days 03:21:36'::interval = '1 mon 29 days 27:21:36'::interval)::text;
```

> The result is _true_. The section [Multiplying or dividing an _interval_ value by a number](./interval-number-multiplication/) simulates and tests the model for how this works in PL/pgSQL code and it examines this unexpected outcome closely.

Reviewer: ```suggestion
The result is _true_. The section [Multiplying or dividing an _interval_ value by a number](./interval-number-multiplication/) simulates and tests the model for how this works in PL/pgSQL code, and examines this unexpected outcome closely.
```

 timestamptz: interval
```

> The _interval_ value that results from subtracting one moment from another (for the _timestamptz_, _timestamp_, or _time_ data types) has, in general, a non-zero value for each of the _dd_ and _ss_ fields of the internal _[&#91;mm, dd, ss&#93;](../interval-representation/)_ representation. The value of the _mm_ field is _always_ zero. The section [The moment-moment overloads of the "-" operator for _timestamptz_, _timestamp_, and _time_](./moment-moment-overloads-of-minus/) explains the algorithm that produces the value and shows that, because it has two fields that have different rules for the semantics of the _interval_-moment overloads of the `+` and `-` operators, this approach for producing an _interval_ value should be avoided. See the section [Custom domain types for specializing the native _interval_ functionality](../custom-interval-domains/) for the recommended alternative approach.

Reviewer: I'm trying to move us away from "see" in favour of "refer to" and its friends, as it's more inclusive and also more accurate.
```suggestion
The _interval_ value that results from subtracting one moment from another (for the _timestamptz_, _timestamp_, or _time_ data types) has, in general, a non-zero value for each of the _dd_ and _ss_ fields of the internal _[&#91;mm, dd, ss&#93;](../interval-representation/)_ representation. The value of the _mm_ field is _always_ zero. The section [The moment-moment overloads of the "-" operator for _timestamptz_, _timestamp_, and _time_](./moment-moment-overloads-of-minus/) explains the algorithm that produces the value and shows that, because it has two fields that have different rules for the semantics of the _interval_-moment overloads of the `+` and `-` operators, this approach for producing an _interval_ value should be avoided. Refer to [Custom domain types for specializing the native _interval_ functionality](../custom-interval-domains/) for the recommended alternative approach.
```
